jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/TypeUtilTest.java

```diff
@@ -254,4 +254,16 @@ public void testToShortName(Class<?> clazz, String shortName)
         assertThat(TypeUtil.toShortName(clazz), is(shortName));
     }
 
+    @Test
+    public void testCeilNextPowerOfTwo()
+    {
+        assertThat(TypeUtil.ceilToNextPowerOfTwo(-1), is(1));
```

Comment:
This is really surprising, I would prefer it throws.

Reply:
technically it is correct as 1 is the next highest power of two for any negative number.   But I don't see that as useful in any way and it is more likely an error, so I will throw.